### PR TITLE
Add i18n guidance re. images & shortcodes + some rework

### LINKED ISSUE #4761: [i18n] Ensure section-local image paths fall back to default lang image
_Originally posted by @chalin in https://github.com/open-telemetry/opentelemetry.io/pull/4741#discussion_r1657789446_

> Hugo is now setup to reuse the default language image -- the exact same file, no duplication (as was the case for static resources a few Hugo versions ago).
> 
> **Images should _not_ be duplicated.** I'd suggest using the default image except where there's too much English text in the image, of course.

### PR DESCRIPTION
**Pull request by chalin**
- Closes #4761
- **Preview**: https://deploy-preview-4790--opentelemetry.netlify.app/docs/contributing/localization/#translation-guidance

### CODE REVIEW

File: content/en/docs/contributing/localization.md [hunk 1]
```diff
@@ -1,45 +1,83 @@
...
+We recommend that you follow the guidance offered in this section when
+translating pages.
+
+### Heading anchors
+
+To ensure that heading anchor targets are uniform across localizations, when
+translating headings:
+
+- Preserve the heading's explicit ID if it has one. [Heading ID syntax] is
+  written after the heading text using syntax like `{ #some-id }`
```
Comment (tiffany76) [inline content/en/docs/contributing/localization.md:25]:
```suggestion
  written after the heading text using syntax like `{ #some-id }`.
```

File: content/en/docs/contributing/localization.md [hunk 2]
```diff
@@ -1,45 +1,83 @@
...
+- Otherwise, explicitly declare a heading ID corresponding to the autogenerated
+  ID of the original English heading.
+
+{{% alert title="Note" %}}
+
+We leave it to the discretion of localization authors to decide if they add an
+explicit heading ID to all headings of translated pages, or whether this is only
+done for known intra-site heading targets -- as reported by the link checker.
```
Comment (tiffany76) [inline content/en/docs/contributing/localization.md:33]:
```suggestion
done for known intra-site heading targets, as reported by the link checker.
```

File: content/en/docs/contributing/localization.md [hunk 3]
```diff
@@ -1,45 +1,83 @@
...
+The former is more uniform and more work, but better supports having
+site-external targets into localization pages, and avoids having to revisit
+previously translated pages as new heading targets are used.
```
Comment (tiffany76) [inline content/en/docs/contributing/localization.md:36]:
```suggestion
The former option is more uniform and more work. It better supports having
site-external targets into localization pages and avoids having to revisit
previously translated pages as new heading targets are used.
```

File: content/en/docs/contributing/localization.md [hunk 4]
```diff
@@ -1,45 +1,83 @@
...
+resources such as images.
 
 {{% alert title="Note" %}}
 
-This repository has a custom render-link hook that Hugo uses to convert
-**absolute link paths to documentation pages** that are of the form
+The OTel website repository has a custom render-link hook that Hugo uses to
+convert **absolute link paths to documentation pages** that are of the form
 `/docs/some-page`, to be locale specific, by prefixing the path with page
 language code when rendering the link. For example, the previous sample path
```
Comment (tiffany76) [inline content/en/docs/contributing/localization.md:54]:
```suggestion
The OTel website repository has a custom render-link hook that Hugo uses to
convert **absolute link paths to documentation pages**. Links of the form
`/docs/some-page` are made locale specific by prefixing the path with the page
language code when rendering the link. For example, the previous sample path
```

File: content/en/docs/contributing/localization.md [hunk 5]
```diff
@@ -1,45 +1,83 @@
...
+localizations. That is, in the generated site folder, Hugo will output a
+_single_ image file that is then shared across localizations.
+
+So as a general rule, _do not_ make copies of image files in your localization
+unless you actually change the image.
+
+### Shortcodes
+
+Some of the base shortcodes contain English text that you might need to localize
+-- this is particularly true of those contain in [layouts/shortcodes/docs].
```
Comment (tiffany76) [inline content/en/docs/contributing/localization.md:71]:
```suggestion
-- this is particularly true of those contained in [layouts/shortcodes/docs].
```

**Review by tiffany76 — APPROVED** [review]
I broke up some of the longer sentences and suggested a few punctuation changes. Otherwise, LGTM!

**Review by svrnm — APPROVED** [review]

**Pull request merged**